Source/DafnyCore/Compilers/Compiler-python.cs

```diff
@@ -751,7 +754,12 @@ private class ClassWriter : IClassWriter {
 
     protected override string TypeName_Companion(Type type, ConcreteSyntaxTree wr, IToken tok, MemberDecl member) {
       type = UserDefinedType.UpcastToMemberEnclosingType(type, member);
```

Comment:
Can we move this inside the else block?

Reply:
No, the updated `type` is also needed in the `if` guard.

Reply:
Are you sure?